test/new-e2e/tests/sysprobe-functional/sysprobe_test.go

```diff
@@ -0,0 +1,128 @@
+	componentsos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/stretchr/testify/require"
+)
+
+type vmSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+var (
+	kitchenDir string
+	testspath  string
+	reporoot   string
+	devMode    = flag.Bool("devmode", false, "run tests in dev mode")
+)
+
+func init() {
+	// Get the absolute path to the test assets directory
+	currDir, _ := os.Getwd()
+
+	reporoot, _ = filepath.Abs(filepath.Join(currDir, "..", "..", "..", ".."))
+	kitchenDir = filepath.Join(reporoot, "test", "kitchen", "site-cookbooks")
+	testspath = filepath.Join(kitchenDir, "dd-system-probe-check", "files", "default", "tests")
+}
+
+func TestVMSuite(t *testing.T) {
+	suiteParams := []e2e.SuiteOption{e2e.WithProvisioner(awshost.ProvisionerNoAgentNoFakeIntake(awshost.WithEC2InstanceOptions(ec2.WithOS(componentsos.WindowsDefault))))}
+	if *devMode {
+		suiteParams = append(suiteParams, e2e.WithDevMode())
+	}
+
+	e2e.Run(t, &vmSuite{}, suiteParams...)
+}
+
+func (v *vmSuite) TestTestSuite() {
+	v.T().Run("Works", v.testExample)
+}
+func (v *vmSuite) testExample(t *testing.T) {
+
+	// get the remote host
+	vm := v.Env().RemoteHost
+
+	err := windows.InstallIIS(vm)
+	require.NoError(t, err)
+	// HEADSUP the paths are windows, but this will execute in linux. So fix the paths
+	t.Log("IIS Installed, continuing")
+
+	t.Log("Creating sites")
+	// figure out where we're being executed from.  These paths should be in
+	// native path separators (i.e. not windows paths if executing in ci/on linux)
+
+	_, srcfile, _, ok := runtime.Caller(0)
+	require.True(t, ok)
+	exPath := filepath.Dir(srcfile)
+
+	sites := []windows.IISSiteDefinition{
+		{
+			Name:        "TestSite1",
+			BindingPort: "*:8081:",
+			AssetsDir:   path.Join(exPath, "assets"),
+		},
+		{
+			Name:        "TestSite2",
+			BindingPort: "*:8082:",
+			AssetsDir:   path.Join(exPath, "assets"),
+		},
+	}
+
+	t.Logf("AssetsDir: %s", sites[0].AssetsDir)
+	err = windows.CreateIISSite(vm, sites)
+	require.NoError(t, err)
+	t.Log("Sites created, continuing")
+
+	rs := windows.NewRemoteSuite(vm, t, "testsuite.exe", testspath)
+	err = rs.FindTestPrograms()
+	require.NoError(t, err)
+
+	err = rs.CreateRemotePaths()
+	require.NoError(t, err)
+
+	err = rs.CopyFiles()
+	require.NoError(t, err)
+
+	// install the agent (just so we can get the driver(s) installed)
+	agentPackage, err := windowsAgent.GetPackageFromEnv()
+	require.NoError(t, err)
+	remoteMSIPath, err := windows.GetTemporaryFile(vm)
+	require.NoError(t, err)
+	t.Log("Getting install package...")
+	err = windows.PutOrDownloadFile(vm, agentPackage.URL, remoteMSIPath)
+	require.NoError(t, err)
+
+	err = windows.InstallMSI(vm, remoteMSIPath, "", "")
+	t.Log("Install complete")
+	require.NoError(t, err)
+
+	// disable the agent, and enable the drivers for testing
+	_, err = vm.Execute("stop-service -force datadogagent")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
As you are using an agent-less instance, should you check that the agent is installed before stopping it ?

Reply:
on line115 the test will fail if the agent is not installed.